### PR TITLE
Update GitHub actions docs for debug/mkl/rocm job

### PR DESCRIPTION
**Pull request by walshmm**
## Proposed changes

This PR updates documentation around Github-Actions and the tests that it runs. More specifically it adds docs for the debug, mkl, and rocm.  It also updates some of intel/AFQMC docs for the pipeline.
It provides some reference information allowing contributors to more easily understand and reproduce the tests ran on their PR's.

## What type(s) of changes does this code introduce?

- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

* Ubuntu 20.04

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
I think some CUDA builds are on nitrogen as well.

**Comment by williamfgc** [comment]
@walshmm @ye-luo is right, you might want to look at the latest `develop` branch (just rebase) as we had recent activity on the self-hosted runners.

**Comment by ye-luo** [comment]
@williamfgc is the doc good now?

File: docs/github_actions.rst [hunk 1]
```diff
@@ -17,64 +17,48 @@ Summary of Test Jobs
...
++---------------------------------------------------+----------+---------------+----------+----------+
+| GCC9-NoMPI-Debug-Real                             | GitHub   | deterministic | 17       | PR/merge |
++---------------------------------------------------+----------+---------------+----------+----------+
+| GCC11-NoMPI-Werror-(Real, Complex)                | GitHub   | deterministic | (60, 60) | PR/merge |
++---------------------------------------------------+----------+---------------+----------+----------+
+| Clang10-NoMPI-ASan-(Real, Complex)                | GitHub   | deterministic | (60, 60) | PR/merge |
++---------------------------------------------------+----------+---------------+----------+----------+
+| Clang10-NoMPI-UBSan-Real                          | GitHub   | deterministic | 60       | PR/merge |
++---------------------------------------------------+----------+---------------+----------+----------+
+| Clang12-NoMPI-Offload-Real                        | GitHub   | unit          | 35       | PR/merge |
```
Comment (williamfgc) [inline docs/github_actions.rst:34]: There are missing configurations recently added, please check with the latest [matrix](https://github.com/QMCPACK/qmcpack/blob/develop/.github/workflows/ci-github-actions.yaml#L23) in configuration file

File: docs/github_actions.rst [hunk 2]
```diff
@@ -441,6 +498,233 @@ gpu-cuda (Clang14Dev-MPI-CUDA-AFQMC-Offload-Real)
 +---------------+-------------------------------------------------------------------------------------------------------------------------------------+
 
 
+gpu-cuda (Intel19-MPI-CUDA-AFQMC-Real-Mixed)
```
Comment (williamfgc) [inline docs/github_actions.rst:501]: Please add a comment or note that for Intel builds, the environment script that ship with the Intel compiler needs to be sourced (e.g. compilervars.sh).

**Review by williamfgc — CHANGES_REQUESTED** [review]

File: docs/github_actions.rst (hunk 2 above)
Comment (walshmm) [reply docs/github_actions.rst:560]: These issues should be resolved, can you confirm? @williamfgc

Comment (ye-luo) [reply docs/github_actions.rst:560]: Ping @williamfgc

File: docs/github_actions.rst [hunk 3]
```diff
@@ -17,64 +17,52 @@ Summary of Test Jobs
...
-+----------------------------------------------+----------+---------------+------+----------+
-| ROCm-Clang13-NoMPI-CUDA2HIP-Complex-Mixed    | nitrogen | deterministic | 6    | manual   |
-+----------------------------------------------+----------+---------------+------+----------+
-| ROCm-Clang13-NoMPI-CUDA2HIP-Complex          | nitrogen | deterministic | 6    | manual   |
-+----------------------------------------------+----------+---------------+------+----------+
++---------------------------------------------------+----------+---------------+----------+----------+
+| Job Name with                                     | Runner   | Tests         | Time     | Trigger  |
+| Build Info                                        | Host     | ctest -L      | min      | event    |
++---------------------------------------------------+----------+---------------+----------+----------+
+| GCC9-MPI-Gcov-(Real*, Complex*)                   | GitHub   | deterministic | (60, 60) | PR/merge |
```
Comment (williamfgc) [inline docs/github_actions.rst:24]: Just bikeshedding, the * * is interpreted as italic for Complex.

**Review by williamfgc — APPROVED** [review]
LGTM, thanks @walshmm

File: docs/github_actions.rst (hunk 3 above)
Comment (ye-luo) [reply docs/github_actions.rst:24]: The rendered readthedocs page seems fine.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**